[hyodol] Step8-2 미션입니다.

bestdevhyo1225 opened the pull request:
1. 기획서 및 설계도
2. 구현 코드 
같이 PR 보냅니다.

bestdevhyo1225 committed:
docs: Create a program design

bestdevhyo1225 committed:
docs: Modify a Image URL

bestdevhyo1225 committed:
feat: Create a server.js file

bestdevhyo1225 committed:
feat: Add a connection event function

bestdevhyo1225 committed:
feat: Add a listen event function

bestdevhyo1225 committed:
feat: Add a data event function (client event)

bestdevhyo1225 committed:
feat: Create a client.js file

bestdevhyo1225 committed:
feat: Add a connection event function (client.js)

bestdevhyo1225 committed:
feat: Add a data event function (client.js)

bestdevhyo1225 committed:
test: Transfer data from client to server

bestdevhyo1225 committed:
test: Transfer data from server to client

bestdevhyo1225 committed:
feat: Create a error event function (server.js)

bestdevhyo1225 committed:
feat: Create a close event function (server.js)

bestdevhyo1225 committed:
feat: Create a close event function (client.js)

bestdevhyo1225 committed:
test: Execute a multi echo client.js

- Enter and Send data
- Receive data from another client

bestdevhyo1225 committed:
test: Execute a multi echo server.js

- Receive data from client
- Send data to all clients

bestdevhyo1225 committed:
feat: Create a utility.js file

- Add readline module (to reuse)

bestdevhyo1225 committed:
feat: Create a app.js file

- It is a file for the user interface

bestdevhyo1225 committed:
test: Simulate the app.js file (user interface)

bestdevhyo1225 committed:
refactor: app.js file

- Seperate function (executeMainPage, executeUserPage)
- Modify variable names and function names

bestdevhyo1225 committed:
refactor: app.js file (second)

- Seperate function (executeSignPage)
- Modify variable names and function names

bestdevhyo1225 committed:
refactor: app.js file (third)

- Separate function
- Modify valible names and function names

bestdevhyo1225 committed:
feat: Create a member_client.js file

- Files to manage members (sign in, sign up)

bestdevhyo1225 committed:
feat: Add a 'sign in' function

bestdevhyo1225 committed:
feat: Create a member_server.js file

- Files to manage members

bestdevhyo1225 committed:
test: Simulate 'sign in' function

bestdevhyo1225 committed:
feat: Add a 'sign up' function

bestdevhyo1225 committed:
test: Simulate 'sign up' function

bestdevhyo1225 committed:
test: Simulate app.js file

bestdevhyo1225 committed:
test: Simulate a  multi echo member client

bestdevhyo1225 committed:
test: Simulate a  multi echo member server

bestdevhyo1225 committed:
feat: Add a node package module

- csv-parser module
- csv-writer module

bestdevhyo1225 committed:
feat: Create a file_manager.js file

- File to manage membership information (member save, lookup)

bestdevhyo1225 committed:
feat: Create a member_information.csv file

- It is a database file

bestdevhyo1225 committed:
fix: Modify a member client bug

bestdevhyo1225 committed:
feat: Add a file save function

bestdevhyo1225 committed:
test: Save a member in the csv file

bestdevhyo1225 committed:
test: Save a test case

bestdevhyo1225 committed:
refactor: Declare a module exports function

bestdevhyo1225 committed:
test: Simulate a sign in function

- Use a setTimeout function for asynchronous processing

bestdevhyo1225 committed:
test: Simulate a sign in function

- Use a pbkdf2 function of crpyto module (for encryption)

bestdevhyo1225 committed:
test: Save a test case

bestdevhyo1225 committed:
feat: Add a setMaxListeners function

bestdevhyo1225 committed:
refactor: Modify a function

bestdevhyo1225 committed:
refactor: Modify and Separate function

bestdevhyo1225 committed:
fix: Modify a bug problem

bestdevhyo1225 committed:
refactor: Redeclare module exports property

bestdevhyo1225 committed:
test: Simulate Chat Service (client part)

bestdevhyo1225 committed:
test: Simulate Chat Service (server part)

bestdevhyo1225 committed:
test: Simulate one chat room service

bestdevhyo1225 committed:
feat: Add a blessed module

bestdevhyo1225 committed:
docs: Modify a design program

bestdevhyo1225 committed:
docs: Final design program document

honux77 reviewed (changes requested):
효돌 왠지 가입 후 로그인이 안 됩니다. 확인해 보세요.

honux77 commented on member_server.js:
```diff
@@ -0,0 +1,37 @@
+        const [type, id, pw] = String(data).split('#');
+        if (type === 'SignIn') {
+            fileManager.signIn(id, pw);
+            setTimeout(() => memberClient.write(fileManager.isMember()), 100);
```
이 부분은 왜 이렇게 했을까요??

bestdevhyo1225 replied:
여러가지 테스트를 해보다가 고쳐야 한다는 것을 잊어버렸습니다. 현재는 문제가 되는 부분을 수정한 상태이며, 코드를 다시 정리한 다음 push 하겠습니다!ㅎㅎ

bestdevhyo1225 committed:
fix: Modify a response function

- return sign up data

bestdevhyo1225 committed:
fix: Modify a sign up function

bestdevhyo1225 committed:
refactor: Modify a some function

- 'search' function returns a promise object
- 'sign' function define a module.exports

bestdevhyo1225 committed:
fix: 'Sign in' bug problem

bestdevhyo1225 committed:
fix: Modify a output log

bestdevhyo1225 committed:
fix: Add a const keyword(sign in and sign up function)

bestdevhyo1225 committed:
docs: Add a program flow image

honux77 commented:
수고하셨어요! 짝짝

honux77 commented:
수고하셨어요! 👍
